lib/codemod/src/transforms/convert-module-to-mdx.js

```diff
@@ -0,0 +1,196 @@
+import recast from 'recast';
+
+// FIXME: duplicate code from @storybook/core start.js
+function isExportStory(key, { includeStories, excludeStories }) {
+  function matches(storyKey, arrayOrRegex) {
+    if (Array.isArray(arrayOrRegex)) {
+      return arrayOrRegex.includes(storyKey);
+    }
+    return storyKey.match(arrayOrRegex);
+  }
+  return (
+    (!includeStories || matches(key, includeStories)) &&
+    (!excludeStories || !matches(key, excludeStories))
+  );
+}
+
+function exportMdx(root, options) {
+  // eslint-disable-next-line no-underscore-dangle
+  const path = root.__paths[0];
+
+  // FIXME: insert the title as markdown after all of the imports
+  return path.node.program.body
+    .map(n => {
+      const { code } = recast.prettyPrint(n, options);
+      if (n.type === 'JSXElement') {
+        return `${code}\n`;
+      }
+      return code;
+    })
+    .join('\n');
+}
+
+function parseIncludeExclude(prop) {
+  const { code } = recast.prettyPrint(prop, {});
+  // eslint-disable-next-line no-eval
+  return eval(code);
+}
+
+/**
+ * Convert a compponent's module story file into an MDX file
+ *
+ * For example:
+ *
+ * ```
+ * input { Button } from './Button';
+ * export default {
+ *   title: 'Button'
+ * }
+ * export const story = () => <Button label="The Button" />;
+ * ```
+ *
+ * Becomes:
+ *
+ * ```
+ * import { Meta, Story } from '@storybook/addon-docs/blocks';
+ * input { Button } from './Button';
+ *
+ * <Meta title='Button' />
+ *
+ * <Story name='story'>
+ *   <Button label="The Button" />
+ * </Story>
+ * ```
+ */
+export default function transformer(file, api) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  // FIXME: save out all the storyFn.story = { ... }
+  const storyKeyToStory = {};
+  // save out includeStories / excludeStories
+  const meta = {};
+
+  function makeAttr(key, val) {
+    return j.jsxAttribute(
+      j.jsxIdentifier(key),
+      val.type === 'Literal' ? val : j.jsxExpressionContainer(val)
+    );
+  }
+
+  function getStoryContents(node) {
+    return node.type === 'ArrowFunctionExpression' && node.body.type === 'JSXElement'
+      ? node.body
+      : j.jsxExpressionContainer(node);
+  }
+
+  function getName(storyKey) {
+    const story = storyKeyToStory[storyKey];
+    if (story) {
+      const name = story.properties.find(prop => prop.key.name === 'name');
+      if (name && name.value.type === 'Literal') {
+        return name.value.value;
+      }
+    }
+    return storyKey;
+  }
+
+  function getStoryAttrs(storyKey) {
+    const attrs = [];
+    const story = storyKeyToStory[storyKey];
+    if (story) {
+      story.properties.forEach(prop => {
+        const { key, value } = prop;
+        if (key.name !== 'name') {
+          attrs.push(makeAttr(key.name, value));
+        }
+      });
+    }
+    return attrs;
+  }
+
+  // 1. If the program does not have `export default { title: '....' }, skip it
```

Comment:
In future would we have ways of setting the title from the component `displayName`?